src/main/java/com/alipay/oceanbase/rpc/stream/ObTableClientQueryStreamResult.java

```diff
@@ -76,10 +80,20 @@ protected ObTableQueryResult execute(ObPair<Long, ObTableParam> partIdWithIndex,
                         if (failedServerList != null) {
                             route.setBlackList(failedServerList);
                         }
-                        subObTable = client
-                            .getTable(indexTableName, partIdWithIndex.getLeft(),
-                                needRefreshTableEntry, client.isTableEntryRefreshIntervalWait(),
-                                route).getRight().getObTable();
+                        if (isNeedRefreshTableGroupName) {
+                            // if input is table group, retry should replace table name
+                            indexTableName = client.tryGetTableNameFromTableGroupCache(client.getTableGroupInverted().get(indexTableName), true);
```

Comment:
We could ignore indexTableName here. It seems that we use real TableName when retry.
Also, we could use 
```
tableName = client.tryGetTableNameFromTableGroupCache(client.getTableGroupInverted().get(indexTableName), isNeedRefreshTableGroupName;
```

Reply:
edit, delete